spec/facter/util/facts/posix/virtual_detector_spec.rb

```diff
@@ -241,6 +241,19 @@
       end
     end
 
+    context 'when smartos-lx' do
+      let(:value) { 'smartos-lx' }
+
+      before do
+        allow(Facter::Resolvers::VirtWhat).to receive(:resolve).with(:vm).and_return('lxc')
```

Comment:
Hi, I’m one of the maintainers for SmartOS. In general, I support this change to facter.

But, I’m not sure returning `lxc` here is the best option. SmartOS LX brand zones are not the same as LXC, and we are not attempting to emulate LXC. This may cause some programs to attempt to use LXC facilities that won’t be present on SmartOS and/or not use facilities present (e.g., `mdata-*` commands). If possible, I would suggest returning `lx` here.

Reply:
Thanks, @bahamat.  This MR intends to do (almost) that (please see [this change](https://github.com/puppetlabs/facter/pull/2492/files#diff-6a457370eec506d706714e7494e9f6debeb0e914837f76224b2eef0ad66a2e69R29).  Currently, `facter` returns `lxc` (passing through the value from `virt-what`), whereas this MR changes it to return `smartos-lx`.  (I figured the `smartos-` prefix would make it clearer than simply `lx`.)

The `spec` (automated test) line you commented on is just mocking the return value of `virt-what`.

Reply:
Ok, that makes sense. Sorry for the confusion.

Reply:
In a6107cb, I renamed the identifier from `smartos-lx` to `illumos-lx` based on [this discussion](https://github.com/TritonDataCenter/illumos-joyent/pull/415#issuecomment-1123304950).